ui/app.py

```
@@ -1,168 +1,191 @@
+uploaded_files = st.file_uploader("Upload a document (optional)", type=['pdf', 'docx', 'csv', 'txt', 'jpg', 'jpeg', 'png', 'gif', 'bmp', 'svg'], accept_multiple_files=True)
+# Start button will be triggered by the chat input
+debug_mode = st.checkbox("Enable debug mode")
+
+if query:
+    st.session_state.messages.append({"role": "user", "content": query})
+
+if st.session_state.messages and st.session_state.messages[-1]["role"] == "user":
+    # Find the user query message
+    user_message = st.session_state.messages[-1]
+    if len(st.session_state.messages) == 1 or st.session_state.messages[-2]["role"] == "user":
+         with st.spinner("The agent is now conducting research... This may take a few minutes."):
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)